### DISCUSSION TITLE
Compressing via blosc library

### DISCUSSION THREAD

**Discussion by eschnett** [ornladios/ADIOS2#3920] [Q&A] [answered]
I wonder how much work it would be to add [blosc](https://www.blosc.org) compression to ADIOS2, and how that would be best implemented? Would this best be handled by plugin? Are there examples for plugins? Does such a plugin exist already?

**Comment by anagainaru** [comment, marked as answer]
Blosc2 is already an operator within ADIOS2 that you can attach to your data. This is an example of using Blosc2:
```c++
        // add operations
        adios2::Operator Blosc2Op =
            adios.DefineOperator("Blosc2Compressor", adios2::ops::LosslessBlosc);

        variable.AddOperation(Blosc2Op, {{adios2::ops::blosc::key::clevel, accuracy},
                                        {adios2::ops::blosc::key::threshold, threshold},
                                        {adios2::ops::blosc::key::doshuffle, doshuffle}});
```
If you are interested in the internal implementation, the operator is defined here:`source/adios2/operator/compress/CompressBlosc.cpp`
